lib/backend/client/hostendpoint.go

```diff
@@ -0,0 +1,52 @@
+package client
+
+import (
+	. "github.com/projectcalico/libcalico/lib/backend/objects"
+)
+
+// HostEndpointInterface has methods to work with Service resources.
+type HostEndpointInterface interface {
+	List(hostname, name *string) ([]HostEndpoint, error)
+	Get(hostname, name string) (*HostEndpoint, error)
+	Create(hostname, name string, data *HostEndpoint) error
+	Update(hostname, name string, data *HostEndpoint) error
+	Delete(hostname, name string) error
+}
+
+// services implements ServicesNamespacer interface
+type hostEndpoints struct {
+	r  *Client
+}
+
+// newServices returns a services
+func newHostEndpoints(c *Client) *hostEndpoints {
+	return &hostEndpoints{c}
+}
+
+// List takes a Metadata, and returns the list of hot endpoints that match that Metadata
+// (wildcarding mising fields)
+func (h *hostEndpoints) List(hostname, name *string) ([]HostEndpoint, error) {
```

Comment:
Yup I was undecided  on this - I think an options struct would be better - especially if we want to (say) add filters etc.  I'll update.